src/current/v24.3/troubleshoot-replication-zones.md

```diff
@@ -0,0 +1,231 @@
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+ALTER TABLE movr.users CONFIGURE ZONE DISCARD;
+~~~
+
+<a name="step2"></a>
+
+### Step 2. Move upward in the inheritance hierarchy as needed
+
+If the previous target schema object looked good, look at its parent schema object at the next level up in [the inheritance hierarchy]({% link {{ page.version.version }}/configure-replication-zones.md%}#how-zone-config-inheritance-works).
+
+This is the new target schema object. Return to [the previous step](#step1) and follow the instructions there.
+
+Continue this process recursively until you either find the misconfigured zone configuration or make it all the way up to [the `default` replication zone]({% link {{ page.version.version }}/configure-replication-zones.md %}#view-the-default-replication-zone) and confirm that all of your schema objects have the expected zone configurations.
+
+## Considerations
+
+<a name="monitoring"></a>
+
+### Monitor for ranges that are under-replicated or violating constraints
+
+Monitor the output of the [critical nodes endpoint]({% link {{ page.version.version }}/monitoring-and-alerting.md %}#critical-nodes-endpoint) to see if you have ranges that are under-replicated, or violating constraints.
+
+Using the range IDs from that endpoint, you can map from range IDs to schema objects as described below. This will give you [a target schema object to start from](#step1).
+
+To monitor for under-replicated range IDs, see [Critical nodes endpoint &gt; Replication status - under-replicated ranges]({% link {{ page.version.version }}/monitoring-and-alerting.md %}#replication-status-under-replicated-ranges).
+
+To monitor for range IDs that are violating constraints, see [Critical nodes endpoint &gt; Replication status - constraint violation]({% link {{ page.version.version }}/monitoring-and-alerting.md %}#replication-status-constraint-violation).
+
+Once you have a range ID, you need to map from that ID to the name of a schema object that you can pass to [`SHOW ZONE CONFIGURATIONS`]({% link {{ page.version.version }}/show-zone-configurations.md %}).
+
+The following example query uses the [`SHOW RANGES`]({% link {{ page.version.version }}/show-ranges.md %}) statement to show, for each range ID, which tables and indexes use that range for their underlying storage. The query assumes the [`movr` schema]({% link {{ page.version.version }}/movr.md %}#the-movr-database) that is loaded by [`cockroach demo`]({% link {{ page.version.version }}/cockroach-demo.md %}), so you'll need to modify it to work with your schema.
+
+XXX: GET HELP WITH THIS QUERY, OUTPUT SEEMS NOT QUITE RIGHT??? (or is it ok?)
```

Comment:
@fqazi and @annrpom , i think this SQL query to map from range IDs to schema object names could be improved, it "kinda works" (tm) but please feel free to suggest improvements

Reply:
(as you can see the output is maybe not the best)

Reply:
this looks fine to me! maybe faizan has other thoughts...

Reply:
I think you can introduce array_aggregation here to make it more readable:

```
 WITH movr_tables AS (SHOW RANGES FROM DATABASE movr WITH TABLES),
                                       movr_indexes AS (SHOW RANGES FROM DATABASE movr WITH INDEXES)
                                   SELECT array_agg(movr_indexes.range_id) as ranges,
                                          movr_tables.table_name,
                                          movr_indexes.index_name
                                     FROM movr_tables, movr_indexes
                                    WHERE movr_tables.range_id = movr_indexes.range_id
                                 GROUP BY  movr_tables.table_name,  movr_indexes.index_name ORDER BY table_name, index_name;
```

WDYT? @annrpom @rmloveland

Reply:
i think it's great @fqazi , agree that it's easier to read, have updated to use this query - thank you! i have also updated the surrounding text to match the new table output